os/arch/arm/src/imxrt/imxrt_clock.c

```diff
@@ -0,0 +1,1129 @@
+#endif
+
+#else
+typedef uint64_t clock_64b_t;
+#endif
+
+/*******************************************************************************
+ * Variables
+ ******************************************************************************/
+
+/* External XTAL (OSC) clock frequency. */
+volatile uint32_t g_xtalFreq;
+/* External RTC XTAL clock frequency. */
+volatile uint32_t g_rtcXtalFreq;
+
+/*******************************************************************************
+ * Prototypes
+ ******************************************************************************/
+
+/*!
+ * @brief Get the periph clock frequency.
+ *
+ * @return Periph clock frequency in Hz.
+ */
+static uint32_t imxrt_clock_getperiphclkfreq(void);
+
+/*******************************************************************************
+ * Code
+ ******************************************************************************/
+
+static uint32_t imxrt_clock_getperiphclkfreq(void)
+{
+    uint32_t freq;
+
+    /* Periph_clk2_clk ---> Periph_clk */
+    if (CCM->CBCDR & CCM_CBCDR_PERIPH_CLK_SEL_MASK) {
+        switch (CCM->CBCMR & CCM_CBCMR_PERIPH_CLK2_SEL_MASK) {
+        /* Pll3_sw_clk ---> Periph_clk2_clk ---> Periph_clk */
+        case CCM_CBCMR_PERIPH_CLK2_SEL(0U):
+            freq = imxrt_clock_getpllfreq(kCLOCK_PllUsb1);
+            break;
+        /* Osc_clk ---> Periph_clk2_clk ---> Periph_clk */
+        case CCM_CBCMR_PERIPH_CLK2_SEL(1U):
+            freq = imxrt_clock_getoscfreq();
+            break;
+        case CCM_CBCMR_PERIPH_CLK2_SEL(2U):
+            freq = imxrt_clock_getpllfreq(kCLOCK_PllSys);
+            break;
+        case CCM_CBCMR_PERIPH_CLK2_SEL(3U):
+        default:
+            freq = 0U;
+            break;
+        }
+
+        freq /= (((CCM->CBCDR & CCM_CBCDR_PERIPH_CLK2_PODF_MASK) >> CCM_CBCDR_PERIPH_CLK2_PODF_SHIFT) + 1U);
+    } else { /* Pre_Periph_clk ---> Periph_clk */
+        switch (CCM->CBCMR & CCM_CBCMR_PRE_PERIPH_CLK_SEL_MASK) {
+        /* PLL2 ---> Pre_Periph_clk ---> Periph_clk */
+        case CCM_CBCMR_PRE_PERIPH_CLK_SEL(0U):
+            freq = imxrt_clock_getpllfreq(kCLOCK_PllSys);
+            break;
+        /* PLL2 PFD2 ---> Pre_Periph_clk ---> Periph_clk */
+        case CCM_CBCMR_PRE_PERIPH_CLK_SEL(1U):
+            freq = imxrt_clock_getsyspfdfreq(kCLOCK_Pfd2);
+            break;
+        /* PLL2 PFD0 ---> Pre_Periph_clk ---> Periph_clk */
+        case CCM_CBCMR_PRE_PERIPH_CLK_SEL(2U):
+            freq = imxrt_clock_getsyspfdfreq(kCLOCK_Pfd0);
+            break;
+        /* PLL1 divided(/2) ---> Pre_Periph_clk ---> Periph_clk */
+        case CCM_CBCMR_PRE_PERIPH_CLK_SEL(3U):
+            freq = imxrt_clock_getpllfreq(kCLOCK_PllArm) /
+                    (((CCM->CACRR & CCM_CACRR_ARM_PODF_MASK) >> CCM_CACRR_ARM_PODF_SHIFT) + 1U);
+            break;
+        default:
+            freq = 0U;
+            break;
+        }
+    }
+
+    return freq;
+}
+
+/*!
+ * brief Initialize the external 24MHz clock.

```

Comment:
Please change format of description to following
/****************************************************************************
 * Name: 
 *
 * Description:
 *
 * Input Parameters:
 *
 * Returned Value:
****************************************************************************/

Reply:
I fixed it
Please check it